space-concordia-robotics/robotics-prototype
# Setup TX2 initialization

# Assignee Section

## Description
Made some code changes and additions to ensure that ROS works properly on the TX2.

- Modified the systemd services and startup scripts that run on the OBC in order for them to work with the user directories of the TX2.
- Created a setup script( not tested very much, likely has issues ), to keep track of installed software and to make setting up OBC go more smoothly in the future.

### Steps for Testing
1) Follow the client section of this [guide](https://docs.google.com/document/d/1N7UnQm65c9YOszqaOxbKguqmpEOwiNORGUgtopzSDnI/edit) for setting up OpenVPN, in order to be able to connect to the Jetson.( Feel free to leave comments in the document to let me know if my instructions suck )
2) Run `ping nividia` to ensure that you are able to communicate with the Jetson.
3) Ssh into the Jetson by using `ssh nvidia@nvidia` and the password `nvidia`( this is also the sudo password ). Add your computer's ip and username to its /etc/hosts file.
3) Run `rosgui` and `startgui`.
4) Run the ros `ping_acknowledgement_client` node to test your computer's connection to ros master on the Jetson.
5) Run `rostopic list` and `rosnode list` on the Jetson to view currently subscribed rostopics and running rosnodes.
5) Open the gui and test run anything else and manually run any other ros packages that you can think of.
6) Run the setup script in a VM or on another user account and ensure it sets everything up properly. Make the sure the user you run it on is named "Nvidia"! ( VM recommended this will install systemd services! ).

**Note:**
For some odd reason, the ros nodes use the Jetson's computer name of 'jetson-1422719019290' as its hostname instead of its username, when using the VPN server and ran from the systemd service. Roscore still uses the hostname 'nvidia', which is why you have to assign its ip to both of these hostnames. This behaviour didn't seem to occur when connecting over ethernet. 

closes #275

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
